### PR TITLE
OP-1840: single doppleganger scenario

### PR DESCRIPTION
**Pull request by TomVasile**
Implements itst11 from https://casperlabs.atlassian.net/wiki/spaces/OPS/pages/1133707277/Casper+Network+Infra+Test+Scenarios.

Bumps finality_threshold_fraction: 1/10 -> 1/3

https://casperlabs.atlassian.net/browse/OP-1840

### CODE REVIEW

**Review by goral09 — APPROVED** [review]
Looks good. Do you think we can extract some of these test utility methods to a file and "source" it in the others?

File: utils/nctl/sh/scenarios/itst11.sh [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+source "$NCTL"/sh/utils/main.sh
+source "$NCTL"/sh/views/utils.sh
+source "$NCTL"/sh/node/svc_"$NCTL_DAEMON_TYPE".sh
+
+# Exit if any of the commands fail.
+set -e
+
+#######################################
+# Runs an integration test that tries to simulate
+# a single doppleganger situation.
```
Comment (afck) [inline utils/nctl/sh/scenarios/itst11.sh:12]:
```suggestion
# a single doppelganger situation.
```
(Also below.)

File: utils/nctl/sh/scenarios/itst11.sh [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+function do_await_era_change() {
+    # allow chain height to grow
+    local ERA_COUNT=${1:-"1"}
+    log_step "awaiting $ERA_COUNT eras…"
+    await_n_eras "$ERA_COUNT"
+}
+
+function create_doppleganger() {
+    local NODE_ID=${1}
+    local DOPPLE_ID=${2}
```
Comment (afck) [inline utils/nctl/sh/scenarios/itst11.sh:105]:
```suggestion
    local DOPPEL_ID=${2}
```

**Review by afck — COMMENTED** [review]
Does this really always pass? I was hoping that our doppelganger detection would at least sometimes prevent an equivocation from actually happening here?

**Comment by TomVasile** [comment]
> Does this really always pass? I was hoping that our doppelganger detection would at least sometimes prevent an equivocation from actually happening here?

From the runs that I did I think so, one of the two nodes would always log as faulty .

**Comment by TomVasile** [comment]
> Looks good. Do you think we can extract some of these test utility methods to a file and "source" it in the others?

I think some probably could, some others are specific to each scenario. Probably a good idea though for a future ticket.

**Comment by afck** [comment]
If one of the two nodes really does see the other in time, it should actually avoid the equivocation and just print:
```
received vertex from a doppelganger.
```
I'd say if that happens, the test should pass, too.

**Comment by TomVasile** [comment]
> If one of the two nodes really does see the other in time, it should actually avoid the equivocation and just print:
> 
> ```
> received vertex from a doppelganger.
> ```
> 
> I'd say if that happens, the test should pass, too.

I added a check for that string as well

**Comment by TomVasile** [comment]
@sacherjj wanted to call the change to `resources/local/chainspec.toml.in` to your attention before I merge. Not sure if you use this file anywhere

File: resources/local/chainspec.toml.in [hunk 3]
```diff
@@ -48,7 +48,7 @@ round_seigniorage_rate = [15_959, 6_204_824_582_392]
 # It is the fraction of validators that would need to equivocate to make two honest nodes see two conflicting blocks as
 # finalized: A higher value F makes it safer to rely on finalized blocks.  It also makes it more difficult to finalize
 # blocks, however, and requires strictly more than (F + 1)/2 validators to be working correctly.
-finality_threshold_fraction = [1, 10]
+finality_threshold_fraction = [1, 3]
```
Comment (goral09) [inline resources/local/chainspec.toml.in:51]: We should/could modify this in `nctl` though.

Comment (TomVasile) [reply resources/local/chainspec.toml.in:51]: we could switch it only for nctl, but if mainnet plans to be run with 1/3 we should probably default to that. I'll let others chime in though , I'll bring it up in standup today

Comment (goral09) [reply resources/local/chainspec.toml.in:51]: I agree as general rule but some tests will be easier to run with different values. :+1:

**Review by afck — APPROVED** [review]

File: resources/local/chainspec.toml.in (hunk 3 above)
Comment (TomVasile) [reply resources/local/chainspec.toml.in:51]: After discussing in standup, we decided to proceed with this as 1/3.

**Review by sacherjj — APPROVED** [review]

**Pull request merged**